### PR TITLE
Dependency graph fix

### PR DESCRIPTION
**Pull request by danpoe**
@lucasccordeiro Please check if that works for you.

Related to #260.

### CODE REVIEW

**Comment by lucasccordeiro** [comment]
@tautschnig: This issue describes the error #260.

**Comment by lucasccordeiro** [comment]
Many thanks, @danpoe. Your patch fixes the problem related to the full-slice option. It would be great if we could get it integrated into CBMC master soon (i.e., implementing @tautschnig suggestion).

File: src/analyses/ai.cpp [hunk 1]
```diff
@@ -499,10 +499,11 @@ void ai_baset::sequential_fixedpoint(
...
   const namespacet &ns)
 {
-  // do each function at least once
+  goto_functionst::function_mapt::const_iterator
+    f_it=goto_functions.function_map.find(goto_functions.entry_point());
 
-  forall_goto_functions(it, goto_functions)
-    fixedpoint(it->second.body, goto_functions, ns);
+  if(f_it!=goto_functions.function_map.end())
+    fixedpoint(f_it->second.body, goto_functions, ns);
```
Comment (tautschnig) [inline src/analyses/ai.cpp]: How does this change relate to the commit message?

Comment (danpoe) [inline src/analyses/ai.cpp]: Ah right. This is a general framework fix as it does not make sense to run the fixpoint on other functions when we don't set the entry state (as discussed with @peterschrammel). Should I make a separate commit or pull request for that?

Comment (tautschnig) [inline src/analyses/ai.cpp]: Apologies for the major delay: Yes, please do move this into a separate commit as it could be contentious for cases where the entry function isn't available (it's not clear to me whether running abstract interpretation considering each function a possible entry point is actually a bad idea).

Comment (danpoe) [inline src/analyses/ai.cpp]: I think it would be a useful new feature actually. The existing code though seemed to be a remaining hack from when bottom could be a state of a reachable location.

**Comment by danpoe** [comment]
@tautschnig I've split up the commits as suggested. If this looks fine I think it can be approved.
